type.go

```diff
@@ -381,12 +401,32 @@ func (t int32Type) AssignValue(dst reflect.Value, src Value) error {
 	default:
 		dst.Set(reflect.ValueOf(v))
 	}
-
 	return nil
 }
 
 func (t int32Type) ConvertValue(val Value, typ Type) (Value, error) {
-	return val, checkTypeKindEqual(t, typ)
+	switch typ.(type) {
+	case *stringType:
+		return convertStringToInt32(val)
+	}
+	switch typ.Kind() {
+	case Boolean:
+		return convertBooleanToInt32(val)
+	case Int32:
```

Comment:
Another option I'm wondering is to throw an error if you try to convert to yourself, to try to prevent people from doing unnecessary conversions (I'm not 100% sure how you are going to use this yet, maybe that's not a feasible option).

Reply:
I don't think this is the right layer to enforce it, this API is exported and could be used in contexts where that's perfectly OK to allow conversions between columns of the same type. The code remains correct since the operation is a no-op in this case.

This code is already in use actually since `parquet.Type.ConvertValue` is the underlying mechanism with which `paruqet.Convert` converts columns that change types between two schemas.